src/Search/Gearman/Command/ImportCommand.php

```diff
@@ -157,6 +172,21 @@ private function iterateSerializeTask(Iterator $items, string $type, $method = '
                     $items->next();
                     continue;
                 }
+
+                if ($item instanceof ArticleVersion || $item instanceof ReviewedPreprint) {
+                    $date = $item->getStatusDate();
+                } elseif ($item instanceof HasPublishedDate) {
+                    $date = $item->getPublishedDate();
+                } else {
+                    $items->next();
+                    continue;
```

Comment:
We should only both collecting the date if a dateFrom is set and we should only skip this item if dateForm is set and no date can be determined from the item.